src/Mafia/Cabal/Version.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Mafia.Cabal.Version
+  ( getCabalVersion
+  , checkCabalVersion
+  ) where
+
+import           Control.Monad.IO.Class (liftIO)
+
+import           Data.Text (Text)
+import qualified Data.Text as T
+
+import           Mafia.Cabal.Process
+import           Mafia.Cabal.Types
+import           Mafia.Package
+import           Mafia.Process
+
+import           P
+
+import           System.IO (IO)
+
+import           X.Control.Monad.Trans.Either (EitherT, left, runEitherT)
+
+
+getCabalVersion :: EitherT CabalError IO Version
+getCabalVersion = do
+  result <- liftIO . runEitherT $ cabal "--version" []
+  case result of
+    Left  _         -> left CabalNotInstalled
+    Right (Out out) ->
+      case parseCabalVersion out of
+        Nothing  -> left (CabalCouldNotParseVersion out)
+        Just ver -> return ver
+
+parseCabalVersion :: Text -> Maybe Version
+parseCabalVersion out =
+  case T.words out of
+    ("cabal-install" : "version" : version : _) -> parseVersion version
+    _                                           -> Nothing
+
+-- TODO not sure if maybe we should have no upper bounds here?
+-- TODO maybe mafia should just install its own cabal?
```

Comment:
Yeah I guess as discussed that would be the next thing. And then mafia can...

![conquer the world](http://bligoo.com/media/users/0/28040/images/pinky_brain.jpg)